src/events/mod.rs

```diff
@@ -390,48 +390,34 @@ impl<'a> BytesDecl<'a> {
     /// Gets xml version, including quotes (' or ")
     pub fn version(&self) -> Result<Cow<[u8]>> {
         // The version *must* be the first thing in the declaration.
-        match self.element.attributes().next() {
-            Some(Err(e)) => Err(e),
-            Some(Ok(Attribute {
-                key: b"version",
-                value: v,
-            })) => Ok(v),
+        match self.element.attributes().with_checks(false).next() {
+            Some(Ok(a)) if a.key == b"version" => Ok(a.value),
+            // first attribute was not "version"
             Some(Ok(a)) => {
                 let found = from_utf8(a.key).map_err(Error::Utf8)?.to_string();
                 Err(Error::XmlDeclWithoutVersion(Some(found)))
             }
+            // error parsing attributes
+            Some(Err(e)) => Err(e),
+            // no attributes
             None => Err(Error::XmlDeclWithoutVersion(None)),
         }
     }
 
     /// Gets xml encoding, including quotes (' or ")
     pub fn encoding(&self) -> Option<Result<Cow<[u8]>>> {
```

Comment:
@tafia, I was discussing with @Mingun whether this signature should be changed to `<Result<Option<Cow<[u8]>>` instead of the way it is currently.  That feels more idiomatic and also more consistent with `version()`.  What do you think?